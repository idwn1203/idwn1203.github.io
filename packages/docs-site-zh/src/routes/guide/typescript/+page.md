---
title: Working with Typescript
---

See 

* [Vite plugin typescript support](/reference/vite-plugin/#Typescript)
* [Default theme typescript support](/reference/default-theme/#Typescript)