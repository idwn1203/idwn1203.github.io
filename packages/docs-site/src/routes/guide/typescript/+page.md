---
title: Working with Typescript
---

See 

* [Vite plugin typescript support](/reference/vite-plugin/#Working-with-typescript)
* [Default theme typescript support](/reference/default-theme/#Working-with-typescript)