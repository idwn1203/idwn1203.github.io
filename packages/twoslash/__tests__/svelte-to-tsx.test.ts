import { readFileSync } from 'node:fs'
import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import { DocumentSnapshot } from 'svelte-language-server/dist/src/plugins/typescript/DocumentSnapshot'
import { parse } from 'svelte/compiler'
import { Document } from 'svelte-language-server/dist/src/lib/documents'

const svelteFilePath = resolve(import.meta.dirname, 'test.svelte')
const svelteCode = readFileSync(svelteFilePath, 'utf-8')

describe('svelte to tsx', () => {
  const doc = DocumentSnapshot.fromDocument(new Document('index.svelte', svelteCode), {
    parse,
    version: '4',
    transformOnTemplateError: false,
    typingsNamespace: '',
  })
  it('tsx code', async () => {
    expect(doc.getFullText()).toMatchSnapshot()
    expect(doc.getOriginalPosition({
      line: 2,
      character: 15,
    })).toMatchInlineSnapshot(`
      {
        "character": 15,
        "line": 1,
      }
    `)
  })
})
