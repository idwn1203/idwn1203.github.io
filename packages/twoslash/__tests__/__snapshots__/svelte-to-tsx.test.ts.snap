// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte to tsx > tsx code 1`] = `
"///<reference types="svelte" />
;
  export const FOO = 'BAR'
;;

import { onMount } from 'svelte'
import FloatingWrapper from '../src/components/FloatingWrapper.svelte'
function render() {

  
  

  let count = 0

  onMount(() => {
    console.log('mounted')
  })
;
async () => {



 { svelteHTML.createElement("button", {  "on:click":() => count++,});
    count;
 }

 { const $$_repparWgnitaolF0C = __sveltets_2_ensureComponent(FloatingWrapper); new $$_repparWgnitaolF0C({ target: __sveltets_2_any(), props: {  "floatingClass":\`custom-floating-class\`,}});   FloatingWrapper}
};
return { props: {} as Record<string, never>, slots: {}, events: {} }}

export default class Index__SvelteComponent_ extends __sveltets_2_createSvelte2TsxComponent(__sveltets_2_with_any_event(render())) {
}"
`;
